app/models/user.rb

```diff
@@ -36,15 +38,30 @@ def forget
     update_attribute :remember_digest, nil
   end
 
-  def authenticated? remember_token
-    return false unless remember_token
+  def authenticated? attribute, token
+    digest = send "#{attribute}_digest"
+    return false unless digest
+
+    BCrypt::Password.new(digest).is_password? token
+  end
 
-    BCrypt::Password.new(remember_digest).is_password? remember_token
+  def activate
+    update_attribute :activated, true
+    update_attribute :activated_at, Time.zone.now
```

Comment:
```suggestion
    update_columns activated: true, activated_at: Time.zone.now
```
chỗ này là tại server update nên ko cần chạy callback hay validate nên mạnh dạn dùng update column nhé cu